sdk/servicebus/Azure.Messaging.ServiceBus/src/Client/ServiceBusClient.cs

```diff
@@ -175,34 +175,28 @@ public ServiceBusSender CreateSender(string queueOrTopicName)
 
             return new ServiceBusSender(
                 entityPath: queueOrTopicName,
-                viaEntityPath: null,
+                options: new ServiceBusSenderOptions(),
                 connection: Connection);
         }
 
         /// <summary>
         /// Creates a <see cref="ServiceBusSender"/> instance that can be used for sending messages to a specific
-        /// queue or topic via a different queue or topic.
+        /// queue or topic.
         /// </summary>
         ///
         /// <param name="queueOrTopicName">The queue or topic to create a <see cref="ServiceBusSender"/> for.</param>
-        /// <param name="viaQueueOrTopicName"></param>
-        ///
-        /// <remarks>
-        /// This is mainly to be used when sending messages in a transaction.
-        /// When messages need to be sent across entities in a single transaction, this can be used to ensure
-        /// all the messages land initially in the same entity/partition for local transactions, and then
-        /// let Service Bus handle transferring the message to the actual destination.
-        /// </remarks>
+        /// <param name="options">The set of <see cref="ServiceBusSenderOptions"/> to use for configuring
+        /// this <see cref="ServiceBusSender"/>.</param>
         ///
         /// <returns>A <see cref="ServiceBusSender"/> scoped to the specified queue or topic.</returns>
-        public ServiceBusSender CreateSender(string queueOrTopicName, string viaQueueOrTopicName)
+        public ServiceBusSender CreateSender(string queueOrTopicName, ServiceBusSenderOptions options)
         {
-            ValidateEntityName(viaQueueOrTopicName);
+            ValidateEntityName(options?.ViaQueueOrTopicName);
             ValidateEntityName(queueOrTopicName);
```

Comment:
I think we shouldn't call `ValidateEntityName` method for `queueOrTopicName` and `ViaQueueOrTopicName` both. Think about if user passes EntityPath with connection string, then one of the above method will fail because we check in `ValidateEntityName` method that if the entity path specified in both(`Connection.EntityPath` and `entityName`) with different value then it will throw an exception.  

```csharp
if (!string.IsNullOrEmpty(Connection.EntityPath) && !string.Equals(entityName, Connection.EntityPath, StringComparison.InvariantCultureIgnoreCase))
{
      throw new ArgumentException(Resources.OnlyOneEntityNameMayBeSpecified);
}
```

Reply:
Yeah, if they pass an entity name in the connection string that means they won't be able to use SendVia, since they are restricted to that one entity. So in this case we would want to throw an exception.

Reply:
Did you forget to update this?

Reply:
Updated!